### PR TITLE
Faster learning and vizualization

### PR DESCRIPTION
**Pull request by ZeeWanderer**
- Add argument parsing to `flappy.py`
  - `--fps` allows to set FPS for faster or slower vizualization. Since Framerate is not used in computing player position the results are correct.
   - `--dump_hitmasks` dumps hitmasks to file end exits. Hitmasks are dumped from random pipe and player but since all sprites of the same type differ only by color it doesen't matter. This is required for `learn.py`
  - add dumped hitmasks file `hitmasks_data.pkl` since it is correct for all cases unless default sprites are changed.
- Add optional arguments to `update_scores` and `dump_qvalues` in `bot.py`
Needed for `learn.py`
  - `dump_qvalues` in `update_scores` allows not to dump q values.
  - `allways_dump` in `dump_qvalues` allows to allways dump qvalues
- Add `learn.py`
Basically this is a lightweight script purely for learning. It lacks any visualization and works a lot faster then `flappy.py`.
  - Usage: `python learn.py [--cmd_out] <number of iterations>`.
  - `--cmd_out` option outputs (iteration | score) to cmd after each iteration.
  - All image parameters like `IMAGE['pipe'][0].det_height()` are substituted by their actual values from default sprites.
  - `hitmasks_data.pkl` is needed because `.convert_alpha()` cannot be called without initing `pygame.display` so the apha channel is not blended which results in incorrect hitmasks.

<details><summary>Illustration</summary>
<p>

image

</p>
</details>

### CODE REVIEW

**Comment by chncyhn** [comment]
Thanks @maximilian578 these look great! I will look into the PRs tonight. 👍

**Comment by ZeeWanderer** [comment]
@chncyhn, by the way why qvalues are dumped each DUMPING_N Iterations? The `qvalues.json` is truncated and written anew each time qvalues are dumped. It would make sense to utilize pygame `QUIT` event and dump qvalues only when game is exited, but then closing interpreter by closing editor or console will result in qvalues not being saved. `atexit.register()` would probably not work either when closing interpreter.

**Comment by chncyhn** [comment]
@maximilian578 honestly it's a matter of convenience. I have forked the Flappy Bird clone from [sourabhv/FlapPyBird](https://github.com/sourabhv/FlapPyBird), and am not aware of `pygame`'s internals. Dumping q-values at every N iteration is just a very simple solution which works. For this project's purposes, the simpler the better.

**Comment by chncyhn** [comment]
@maximilian578 really cool stuff! I've just changed a couple of things in naming. Merging! 🎉
